<a href="https://colab.research.google.com/github/dsreliete/Womakerscode-DA-desafio_squad/blob/main/Desafio6SquadMaryJackson.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import matplotlib.pyplot as plt
import scipy.stats as stats
import pandas as pd
import numpy as np
import seaborn as sns
import statsmodels.api as sm

In [2]:
url = "https://raw.githubusercontent.com/dsreliete/Womakerscode-DA-desafio_squad/refs/heads/main/Heart_disease_statlog.csv"
df = pd.read_csv(url)
df

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  \
0     70    1   3       130   322    0        2      109      0      2.4   
1     67    0   2       115   564    0        2      160      0      1.6   
2     57    1   1       124   261    0        0      141      0      0.3   
3     64    1   3       128   263    0        0      105      1      0.2   
4     74    0   1       120   269    0        2      121      1      0.2   
..   ...  ...  ..       ...   ...  ...      ...      ...    ...      ...   
265   52    1   2       172   199    1        0      162      0      0.5   
266   44    1   1       120   263    0        0      173      0      0.0   
267   56    0   1       140   294    0        2      153      0      1.3   
268   57    1   3       140   192    0        0      148      0      0.4   
269   67    1   3       160   286    0        2      108      1      1.5   

     slope  ca  thal  target  
0        1   3     1       1  
1        1   0     3       0  
2        0   0     3       1  
3        1   1     3       0  
4        0   1     1       0  
..     ...  ..   ...     ...  
265      0   0     3       0  
266      0   0     3       0  
267      1   0     1       0  
268      1   0     2       0  
269      1   3     1       1  

[270 rows x 14 columns]

**Classificação de Doenças Cardíacas**

O objetivo desse conjunto de dados é realizar uma tarefa de classificação, onde se deve prever se uma pessoa tem ou não tem doença cardíaca. A variável de saída (o que queremos prever) é binária, sendo:
0: a pessoa não sofre de doença cardíaca.
1: a pessoa sofre de doença cardíaca.

**Dicionário de dados**
Este banco de dados contém 13 atributos e uma variável de destino. Possui 8 valores nominais e 5 valores numéricos. A descrição detalhada de todos esses recursos é a seguinte:

* Age: idade dos pacientes em anos
* Sex: (Masculino: 1; Feminino: 0)
* cp: Tipo de dor torácica sentida pelo paciente. Este termo é categorizado em 4
categorias.
* 0 angina típica,
* 1 angina atípica,
* 2 dor não anginosa,
* 3 assintomática
* trestbps: nível de pressão arterial do paciente no modo de repouso em mm/HG
* chol: colesterol sérico em mg/dl
* fbs: Níveis de açúcar no sangue em jejum > 120 mg/dl representa 1 em caso de verdadeiro e 0 como falso (Nominal) ●
restecg: O resultado do eletrocardiograma em repouso é representado em 3 valores distintos
* 0: Normal
* 1: com anormalidade da onda ST-T (inversões da onda T e/ou elevação ou depressão do ST > 0,05 mV)
* 2: mostrando provável ou definitiva hipertrofia ventricular esquerda por Critérios de Estes ●
thalach: frequência cardíaca máxima alcançada
* exang: Angina induzida pelo exercício
* 0 retratando Não
* 1 retratando Sim
* oldpeak: Depressão do ST induzida pelo exercício em relação ao estado de repouso
* slope: segmento ST medido em termos de inclinação durante o pico do exercício 0: inclinação ascendente;
* 1: plano;
* 2: inclinação descendente
* ca: O número de vasos principais (0–3) (nominal)
* thal: Um distúrbio sanguíneo chamado talassemia
* 0: NULO
* 1: fluxo sanguíneo normal
* 2: defeito fixo (sem fluxo sanguíneo em alguma parte do coração)
* 3: defeito reversível (um fluxo sanguíneo é observado, mas não é normal (nominal)
* target: É a variável alvo que temos que prever 1 significa que o paciente sofre de
doença cardíaca e 0 significa que o paciente é normal.

**Perguntas:**

1- Faça uma análise exploratória dos dados, observando as principais variáveis e sua relação com a variável target.


2 - Construa um modelo de regressão logística para classificar se o indivíduo sofre de doença cardíaca ou não.


3 - Analise o resultado da regressão logística e plote a matriz de confusão.